packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -218,6 +238,60 @@ export class CohereChatLanguageModel implements LanguageModelV1 {
                 return;
               }
 
+              case 'tool-calls-chunk': {
+                if (value.tool_call_delta) {
+                  if (value.tool_call_delta.index != toolCalls.length - 1) {
+                    const toolCallId = generateId();
+
+                    toolCalls.push({
+                      toolCallId,
+                      toolName: '',
+                    });
+                  }
+
+                  if (value.tool_call_delta.name) {
+                    const { index } = value.tool_call_delta;
+                    toolCalls[index].toolName = value.tool_call_delta.name;
+
+                    controller.enqueue({
+                      type: 'tool-call-delta',
+                      toolCallType: 'function',
+                      toolCallId: toolCalls[index].toolCallId,
+                      toolName: toolCalls[index].toolName,
+                      argsTextDelta: '',
+                    });
+                  } else if (value.tool_call_delta.parameters) {
+                    const { index } = value.tool_call_delta;
+
+                    controller.enqueue({
+                      type: 'tool-call-delta',
+                      toolCallType: 'function',
+                      toolCallId: toolCalls[index].toolCallId,
+                      toolName: toolCalls[index].toolName,
+                      argsTextDelta: value.tool_call_delta.parameters,
+                    });
+                  }
+                }
+                return;
+              }
+
+              case 'tool-calls-generation': {
+                let index = 0;
+
+                for (const toolCall of value.tool_calls) {
+                  controller.enqueue({
+                    type: 'tool-call',
+                    toolCallId: toolCalls[index].toolCallId,
+                    toolName: toolCalls[index].toolName,
+                    toolCallType: 'function',
+                    args: JSON.stringify(toolCall.parameters),
+                  });
+
+                  index += 1;
+                }
```

Comment:
recommend switching to a regular for loop since you need an index

Reply:
updated!